spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+        public string projectId { get; set; }
+        [Value(1, HelpText = "The ID of the instance where the sample database resides.", Required = true)]
+        public string instanceId { get; set; }
+        [Value(2, HelpText = "The ID of the database where the sample database resides.", Required = true)]
+        public string databaseId { get; set; }
+    }
+
+    [Verb("readWriteWithTransaction", HelpText = "Update data in the sample Spanner database table using a read-write transaction.")]
+    class ReadWriteWithTransactionOptions
+    {
+        [Value(0, HelpText = "The project ID of the project to use when managing Spanner resources.", Required = true)]
+        public string projectId { get; set; }
+        [Value(1, HelpText = "The ID of the instance where the sample database resides.", Required = true)]
+        public string instanceId { get; set; }
+        [Value(2, HelpText = "The ID of the database where the sample database resides.", Required = true)]
+        public string databaseId { get; set; }
+    }
+
+    public class Program
+    {
+        public static async Task<object> CreateSampleDatabaseAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START create_database]
+            // Create client.
+            DatabaseAdminClient databaseAdminClient =
+                await DatabaseAdminClient.CreateAsync();
+            // Initialize request argument(s).
+            InstanceName parent = new InstanceName(projectId, instanceId);
+            string createStatement = $"CREATE DATABASE `{databaseId}`";
+            // Make the request.
+            var createDatabaseRequest =
+                await databaseAdminClient.CreateDatabaseAsync(parent,
+                createStatement);
+            // Poll until the returned long-running operation is complete.
+            var createDatabaseResponse =
+                await createDatabaseRequest.PollUntilCompletedAsync();
+            // Display the operation result.
+            Console.WriteLine("Database create operation state:"
+                + $" {createDatabaseResponse.Result.State}");
+            // Define create table statements for two tables.
+            string[] createTableStatements = {
+               @"CREATE TABLE Singers (
+                     SingerId INT64 NOT NULL,
+                     FirstName    STRING(1024),
+                     LastName STRING(1024),
+                     ComposerInfo   BYTES(MAX)
+                 ) PRIMARY KEY (SingerId)",
+                 @"CREATE TABLE Albums (
+                     SingerId     INT64 NOT NULL,
+                     AlbumId      INT64 NOT NULL,
+                     AlbumTitle   STRING(MAX)
+                 ) PRIMARY KEY (SingerId, AlbumId),
+                 INTERLEAVE IN PARENT Singers ON DELETE CASCADE"};
+            // Make the request.
+            var response =
+                await databaseAdminClient.UpdateDatabaseDdlAsync(

```

Comment:
whats wrong with using a spanner ddl command for this?
you know you can create a connection string that only has projectid + instanceid (no db name).